File: 479/B-two-gram/solution.py
n = int(input())
s = input()

best_count = 0
for i in range(n - 1):
    substr = s[i:i+2]
    pos = s.find(substr, 0)
    count = 0
    while pos != -1:
        count += 1
        pos = s.find(substr, pos+1)
    if count > best_count:
        ans = substr
        best_count = count
print(ans)
    

File: 479/B-two-gram/solution.cpp
#include <iostream>

using namespace std;

int main() {
    int n, bestCount = 0;
    cin >> n;
    string s, ans;
    cin >> s;

    for (int i = 0; i < n - 1; i++) {
        string substring = s.substr(i, 2);
        size_t pos = s.find(substring, 0);
        int count = 0;
        while(pos != string::npos)
        {
            count++;
            pos = s.find(substring, pos+1);
        }
        if (bestCount < count) {
            bestCount = count;
            ans = substring;
        }
    }
    cout << ans << endl;
    return 0;
}
